### PR TITLE
Fix #8194 create url cache

### PR DESCRIPTION
**Pull request by klimov-paul**
Fix #8194 create url cache

### CODE REVIEW

File: framework/web/UrlManager.php [hunk 1]
```diff
@@ -309,18 +310,38 @@ public function createUrl($params)
...
+                        $this->_createUrlRuleMatches[$cacheKey][] = $rule;
+                        break;
+                    }
+                }
+            }
+
+            if ($url !== false) {
+                if (strpos($url, '://') !== false) {
+                    if ($baseUrl !== '' && ($pos = strpos($url, '/', 8)) !== false) {
+                        return substr($url, 0, $pos) . $baseUrl . substr($url, $pos);
```
Comment (samdark) [inline framework/web/UrlManager.php]: What about non-English URLs?

Comment (klimov-paul) [inline framework/web/UrlManager.php]: This is not my addition. This code is just moved out from `foreach`.

Comment (samdark) [inline framework/web/UrlManager.php]: OK.

File: framework/web/UrlManager.php [hunk 2]
```diff
@@ -309,18 +310,38 @@ public function createUrl($params)
...
                     } else {
-                        return "$baseUrl/{$url}{$anchor}";
+                        return $url . $baseUrl . $anchor;
```
Comment (qiangxue) [inline framework/web/UrlManager.php]: Is this change correct?

Comment (klimov-paul) [inline framework/web/UrlManager.php]: This is not a change. GIT just compose a 'smart' diff, so it is look that way.
This is actually a former line 319 (see above). It just have been moved.
Original line now is line 344 - it is intact.

**Comment by qiangxue** [comment]
Thanks! https://github.com/yiisoft/yii2/commit/ebc089e6c26ba85642e5b32eaae017c6b637be79

**Pull request closed without merging**